restock.lic

```diff
@@ -1,58 +1,65 @@
 #   Documentation: https://elanthipedia.play.net/Lich_script_repository#restock
-
 custom_require.call(%w[common common-items common-money])
-
 class Restock
   include DRC
   include DRCM
   include DRCI
-
   def initialize
     setup
-
     items = parse_restockable_items
+
+    custom_loc_items = items.select{ |k| k['hometown'] }
+
+    items -= custom_loc_items
+    restock_items(items)
+
+    unique_hometowns = custom_loc_items.map{ |k| "#{k['hometown']}" }.uniq
+    unique_hometowns.each do |hometown|
+      custom_list = custom_loc_items.select{ |k| k['hometown'].eql?(hometown) }
+      custom_loc_items -= custom_list
+      restock_items(custom_list)
+    end
+  end
+  def restock_items(item_list)
     items_to_restock = []
     coin_needed = 0
 
-    items.each do |item|
+    item_list.each do |item|
       remaining = if item['stackable']
                     count_stackable_item(item['name'])
                   else
                     count_nonstackable_item(item['name'])
                   end
-
       next unless remaining < item['quantity']
       num_needed = item['quantity'] - remaining
       buy_num = (num_needed / item['size'].to_f).ceil
       coin_needed += buy_num * item['price']
       item['buy_num'] = buy_num
       items_to_restock.push(item)
     end
-
-    ensure_copper_on_hand(coin_needed, @settings) if coin_needed > 0
-
+    town = @settings.hometown
```

Comment:
❓ What would you think about having `town` be a second parameter to `restock_items`? Then line 14 would pass in `@settings.hometown` and line 20 would pass in `hometown`

Reply:
Yup, good change. Thank you.